cmd/pj-rehearse/main.go

```diff
@@ -57,16 +56,15 @@ type options struct {
 }
 
 func gatherOptions() (options, error) {
-	o := options{}
+	o := options{kubernetesOptions: flagutil.KubernetesOptions{NOInClusterConfigDefault: true}}
 	fs := flag.CommandLine
 
 	fs.BoolVar(&o.dryRun, "dry-run", true, "Whether to actually submit rehearsal jobs to Prow")
 
 	fs.StringVar(&o.debugLogPath, "debug-log", "", "Alternate file for debug output, defaults to stderr")
 	fs.StringVar(&o.releaseRepoPath, "candidate-path", "", "Path to a openshift/release working copy with a revision to be tested")
 	fs.StringVar(&o.prowjobKubeconfig, "prowjob-kubeconfig", "", "Path to the prowjob kubeconfig. If unset, default kubeconfig will be used for prowjobs.")
-	fs.StringVar(&o.kubeconfigDir, "kubeconfig-dir", "", "Path to the directory containing kubeconfig files to use for CLI requests.")
```

Comment:
Do we need to worry about retaining this existing option and then migrate after your changes have been released, before then ultimately deleting this option? I guess what I'm asking is, is there a chance that removing this option before migrating could break something? Or do we not need to worry about that here?

Reply:
`--kubeconfig-dir` is used by `flagutil.KubernetesOptions` too.

https://github.com/kubernetes/test-infra/blob/4718430582c40c0ffe51a48f86b499cda5556714/prow/flagutil/kubernetes_cluster_clients.go#L175

This PR uses the code from upstream directly instead of reinventing it in ci-tools.